api/types/appserver.go

```diff
@@ -308,3 +309,64 @@ func (s AppServers) Less(i, j int) bool {
 
 // Swap swaps two app servers.
 func (s AppServers) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+// SortByCustom custom sorts by given sort criteria.
+func (s AppServers) SortByCustom(sortBy *SortBy) error {
+	if sortBy == nil {
+		return nil
+	}
+
+	// We assume sorting by type AppServer, we are really
+	// wanting to sort its contained resource Application.
+	isDesc := sortBy.IsDesc
+	switch sortBy.Field {
+	case ResourceMetadataName:
+		sort.Slice(s, func(i, j int) bool {
```

Comment:
Should we use `sort.SliceStable` here and everywhere else? From a UX perspective I think it would be a little funny if two equal elements swapped locations when going from page 1 to page 2 and then back to page 1

>func Slice(x interface{}, less func(i, j int) bool)
Slice sorts the slice x given the provided less function. It panics if x is not a slice.
The sort is not guaranteed to be stable: equal elements may be reversed from their original order. For a stable sort, use SliceStable.
The less function must satisfy the same requirements as the Interface type's Less method.
https://pkg.go.dev/sort#Slice

Reply:
The original resource slice is built up in backend order, right? Is that guaranteed to be consistent over time?

Reply:
We append the next page result to the previous page result. Refreshing might result in it being swapped but not sure this matters too much?

Reply:
Just asking because I personally find it distracting/annoying when a data table switches two equivalent rows on a refresh. Especially if I am trying to find something in the table and all of sudden things aren't where they were before. 

 If the backend didn't give us some consistent order wouldn't that break pagination?

Reply:
> If the backend didn't give us some consistent order wouldn't that break pagination?

ah you're right b/c we sort entire list before we filter, the sorted has to always be in same order. revising to `sort.SliceStable`

Reply:
> I personally find it distracting/annoying when a data table switches two equivalent rows on a refresh

If it happened across a page border you could actually completely miss a row!